File: hash_performance/experiment1.py
#!/usr/bin/env python

import sys
import random
import timeit
import __builtin__




ITERATIONS=10000
hash = {} 
target_index = 0
y = []

for exponent in range(1,20): 
	size = 2**exponent
	hash = {} 
	target_index = 'qwe'
	for n in range(1,size):
		index = random.random()
		hash[index] = random.random()
		if n == size/2:
			target_index = index
	__builtin__.__dict__.update(locals())
	s = "val = hash[target_index]"
	t = timeit.Timer(s)
	print "retrieving an element from a hash with %d elements %d times tooks %.2f msec" % (size, ITERATIONS, 1000 * t.timeit(number=ITERATIONS))
	y.append("%.2f" % (1000 * t.timeit(number=ITERATIONS)) )
	

f = open("values.python-%s" %  sys.version[:5], 'w')
for i in y:
	f.write("%s\n" % i)

File: hash_performance/experiment1.js
(function() {
  var Reporter, b, exponent, fs, hash, index, iterations, n, reporter, size, stream, target_index, timeit;
  fs = require('fs');
  timeit = require('timeit');
  b = require('b');
  iterations = 10000;
  Reporter = (function() {
    function Reporter() {
      this.data = [];
    }
    Reporter.prototype.report = function(name, result, iterations) {
      console.log("retrieving an element from a hash with " + size + " elements " + iterations + " times tooks " + result + " ms");
      return this.data.push(result);
    };
    return Reporter;
  })();
  reporter = new Reporter;
  for (exponent = 1; exponent <= 20; exponent++) {
    size = Math.pow(2, exponent);
    hash = {};
    target_index = 0;
    for (n = 1; 1 <= size ? n <= size : n >= size; 1 <= size ? n++ : n--) {
      index = Math.random();
      hash[index] = Math.random();
      if (n === size / 2) {
        target_index = index;
      }
    }
    b('Read Test').reporter(reporter).run(iterations, function() {
      var val;
      return val = hash[target_index];
    });
  }
  stream = fs.createWriteStream("values.node-" + process.versions.node);
  stream.once('open', function(fd) {
    var val, _i, _len, _ref, _results;
    _ref = reporter.data;
    _results = [];
    for (_i = 0, _len = _ref.length; _i < _len; _i++) {
      val = _ref[_i];
      _results.push(stream.write("" + val + "\n"));
    }
    return _results;
  });
}).call(this);
